examples/example_perturbation_expression_prediction.py

```diff
@@ -103,38 +120,62 @@
-        )  # TODO: Once PR 381 is merged, use the new load_local_dataset function
+            "replogle_k562_essential_perturbpredict",
+            config_path=Path(tmp_cfg_path),
+        )
+
+    # Optional: validate the dataset
+    dataset.validate()
+    # This generates a sample model anndata file. In applications,
+    # this should contain the model predictions and should be provided by the user.
+    model_adata = generate_random_model_predictions(
+        dataset.adata.shape[0], dataset.adata.shape[1]
+    )
 
     # Choose approach based on flag
     if args.save_inputs:
-        print("Using save/load approach...")
+        logger.info("Using save/load approach...")
         # Save and load dataset task inputs
         task_inputs_dir = dataset.store_task_inputs()
-        print(f"Task inputs saved to: {task_inputs_dir}")
+        logger.info(f"Task inputs saved to: {task_inputs_dir}")
         task_input = load_perturbation_task_input_from_saved_files(task_inputs_dir)
-        print("Task inputs loaded from saved files")
+        logger.info("Task inputs loaded from saved files")
+
+        # Update with the model ordering of the genes and of the cells
+        task_input.gene_index = model_adata.var.index
+        task_input.cell_index = model_adata.obs.index
     else:
-        print("Creating task input directly from dataset...")
-        # Create task input directly from dataset
+        logger.info("Creating task input directly from dataset...")
+        # Create task input directly from dataset with separate fields
         task_input = PerturbationExpressionPredictionTaskInput(
+            adata=dataset.control_matched_adata,
+            target_conditions_dict=dataset.target_conditions_dict,
             de_results=dataset.de_results,
-            var_index=dataset.control_matched_adata.var.index,
-            masked_adata_obs=dataset.control_matched_adata.obs,
-            target_conditions_to_save=dataset.target_conditions_to_save,
-            row_index=dataset.adata.obs.index,
+            gene_index=model_adata.var.index,
+            cell_index=model_adata.obs.index,
         )
-
-    # Generate random model output
-    model_output: CellRepresentation = np.random.rand(
-        dataset.adata.shape[0], dataset.adata.shape[1]
-    )
+    # Convert model adata to cell representation
+    model_output = model_adata.X
 
     # Run task
-    task = PerturbationExpressionPredictionTask(metric=args.metric)
-    metrics_dict = task.run(model_output, task_input)
+    task = PerturbationExpressionPredictionTask(
+        condition_key=dataset.condition_key, control_name=dataset.control_name
+    )
+    metrics_dict = task.run(cell_representation=model_output, task_input=task_input)
+    logger.info("Model metrics:")
     print_metrics_summary(metrics_dict)
+
+    # # Compute baseline -- pseudocode -- this throws an error because of non-log-normalized data
```

Comment:
@polinabinder1 This doesn't work, can you update so we have a working version in our script? Thank you.